services/web-app/utils/auth-checkers/validUserAuthorizationChecker.js

```diff
@@ -0,0 +1,22 @@
+import { retrieveUser } from '../retrieveUser'
+
+/*
+ * Copyright IBM Corp. 2021, 2021
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+/**
+ * Determines if user is authorized based on IBM email
+ * @param {GetServerSidePropsContext} context - getServerSideProps context object
+ * @returns {boolean} boolean indicating whether user is authorized to view resource or not
```

Comment:
minor: Technically this returns either `undefined`, `true`, or `false` because of the optional chaining. This will matter more once we move this service to typescript, so in preparation for that, I'd recommend either using `!!` or changing the conditional to `if (user && user.email) {`

Reply:
Done! , went with the !! approach